refactor(core): Add `ngDevMode` guards and new sanitization error codes

SkyZeroZx opened the pull request:
Adds new runtime sanitization error codes. Adds `ngDevMode`  guards around
error message strings to ensure detailed diagnostics are included only
in development mode. This allows production builds to tree-shake

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

JeanMeche commented on packages/core/src/sanitization/html_sanitizer.ts:
```diff
@@ -314,7 +316,11 @@ export function _sanitizeHtml(defaultDoc: any, unsafeHtmlInput: string): Trusted
     do {
       if (mXSSAttempts === 0) {
-        throw new Error('Failed to sanitize html because the input is unstable');
+        throw new Error(
```
Those should be converted to `RuntimeError`. And error message without message nor code really isn't actionnable in a prod bundle.

JeanMeche commented on packages/core/src/errors.ts:
```diff
@@ -134,6 +134,9 @@ export const enum RuntimeErrorCode {
   MISSING_DIRECTIVE_DEFINITION = 916,
   NO_COMPONENT_FACTORY_FOUND = 917,
   EXTERNAL_RESOURCE_LOADING_FAILED = 918,
+  HTML_SANITIZATION_UNSTABLE = 919,
```
This code already conflicts with #65966.

JeanMeche reviewed (approved):
reviewed-for: public-api, core

josephperrott reviewed (approved):
LGTM

Reviewed-for: fw-security

jelbourn reviewed (approved):
LGTM

Reviewed-for: fw-security

JeanMeche commented:
We need a reformating because of #66056.

SkyZeroZx committed:
refactor(core): Add `ngDevMode` guards and new sanitization error codes

Adds new runtime sanitization error codes. Adds `ngDevMode` guards around
error message strings to ensure detailed diagnostics are included only
in development mode. This allows production builds to tree-shake verbose error descriptions, reducing bundle size.

crisbeto commented:
I'm rolling this change back in #66291, because it broke a test internally. Feel free to reopen, but we'll have to be more careful when landing it next time.

SkyZeroZx commented:
> I'm rolling this change back in #66291, because it broke a test internally. Feel free to reopen, but we'll have to be more careful when landing it next time.

Should I make any changes on my end, or is it enough to reopen it as it was?
